src/zjs_callbacks.c

```diff
@@ -90,6 +94,21 @@ static int zjs_ringbuf_error_count = 0;
 static int zjs_ringbuf_error_max = 0;
 static int zjs_ringbuf_last_error = 0;
 
+#ifdef DEBUG_BUILD
+static void set_info_string(char **str, const char *file, const char *func)
+{
+    uint32_t file_len = strlen(file);
+    uint32_t func_len = strlen(func);
+    *str = zjs_malloc(file_len + func_len + 4);
+    strncpy(*str, file, file_len);
+    (*str)[file_len] = ':';
+    strncpy(*str + file_len + 1, func, func_len);
+    (*str)[file_len + func_len + 1] = '(';
+    (*str)[file_len + func_len + 2] = ')';
+    (*str)[file_len + func_len + 3] = '\0';
```

Comment:
Check malloc rval. Probably still set the *str to NULL in that case, just make sure you handle that elsewhere. I think printf("%s", NULL) works okay, just ugly but okay for debug in this rare case.

For readability, I suggest declare a char * and then assign *str to it at the end. Although after the following it won't matter as much.

And then replace most of this with:
snprintf("%s:%s()", file, func);
... FTW!